lib/spack/spack/test/cmd/compiler.py

```diff
@@ -247,3 +248,76 @@ def test_compiler_list_empty(no_compilers_yaml, working_env, compilers_dir):
     out = compiler("list")
     assert not out
     assert compiler.returncode == 0
+
+
+@pytest.mark.parametrize(
+    "external,expected",
+    [
+        (
+            {
+                "spec": "gcc@=7.7.7 os=foobar target=x86_64",
+                "prefix": "/path/to/fake",
+                "modules": ["gcc/7.7.7", "foobar"],
+                "extra_attributes": {
+                    "paths": {
+                        "cc": "/path/to/fake/gcc",
+                        "cxx": "/path/to/fake/g++",
+                        "fc": "/path/to/fake/gfortran",
+                        "f77": "/path/to/fake/gfortran",
+                    },
+                    "flags": {"fflags": "-ffree-form"},
+                },
+            },
+            """gcc@7.7.7:
+\tpaths:
+\t\tcc = /path/to/fake/gcc
+\t\tcxx = /path/to/fake/g++
+\t\tf77 = /path/to/fake/gfortran
+\t\tfc = /path/to/fake/gfortran
+\tflags:
+\t\tfflags = ['-ffree-form']
+\tmodules  = ['gcc/7.7.7', 'foobar']
+\toperating system  = foobar
```

Comment:
Do we need to put tabs in there? Wondering cause as far as I remember tabs are not allowed in YAML, and this makes the literal a bit more obfuscated.

Reply:
Oh, that's because we are checking `compiler("info")`. It's not worth blocking this on a nitpick comment like that, but we might want to simplify this test later, and avoid testing a commend output.